headers/types/dungeon_mode/dungeon_mode.h

```diff
@@ -782,6 +820,140 @@ struct tile {
 };
 ASSERT_SIZE(struct tile, 20);
 
+// Struct that seems to hold data related to the map, the camera and the touchscreen numbers
+struct display_data {
+    struct position camera_pos; // 0x0: Position of the tile the camera is currently pointing to
+    // 0x4: Copy of 0x0. Used to detect changes in the camera's position.
+    struct position camera_pos_mirror;
+    // 0x8: Pixel position of the tile the camera is currently pointing to
+    struct position camera_pixel_pos;
+    struct position camera_pixel_pos_mirror; // 0xC: Mirror of 0x8
+    // 0x10: Entity currently being pointed by the camera, mostly used to
+    // control rendering of the dungeon, GUI, minimap, etc.
+    struct entity* camera_target;
+    undefined field_0x14;
+    undefined field_0x15;
+    undefined field_0x16;
+    undefined field_0x17;
+    undefined field_0x18;
+    undefined field_0x19;
+    undefined field_0x1A;
+    undefined field_0x1B;
+    undefined field_0x1C;
+    undefined field_0x1D;
+    undefined field_0x1E;
+    undefined field_0x1F;
+    undefined field_0x20;
+    // 0x21: Same as floor_properties::darkness_level
+    // Affects the number of map tiles around the player's position that get marked as
+    // "visited" while exploring, as well as how far away you can see enemies under non-illuminated
+    // conditions (outside of this range, enemies will not be visible on screen).
+    uint8_t darkness_level;
```

Comment:
Coming up with a proper name for this one is complicated. Because it also affects whether the dungeon will graphically display darkness or not (darkness is displayed if the value is > 0), as well as visibility in rooms (which is better when the value is 0 and worse if it's > 0).
This change also makes it match `floor_properties::darkness_level`, so it might be a better name?

Also, this is the field that used to be in `dungeon.h`, it's actually part of a separate struct so I moved it here.

Reply:
I felt okay with the name `floor_properties::darkness_level` because it was an enum, where the labels made it clear what the values meant. If we're changing this to a plain `uint8_t` (which I do think is fair because unnatural values still work just fine), I think the name becomes more important.

I'm still partial to the name `visibility_range` as was previously in the dungeon struct (and I think we should change the field in `floor_properties` to match). It's more compatible with the _numeric_ value of the field, and 0 should be understood to be a special case anyway (infinite visibility, which also disables the "darkness" effect that's otherwise present by default).

Reply:
Hmmm, alright. Makes sense.